### PR TITLE
fix: write better result/lexure code based on review

### PR DESCRIPTION
**Pull request by samfundev**

### CODE REVIEW

File: src/lib/parsers/Args.ts [hunk 1]
```diff
@@ -45,16 +45,16 @@ export class Args {
 	/**
 	 * The internal Lexure parser.
 	 */
-	protected readonly parser: Lexure.ArgumentStream;
+	protected readonly parser: ArgumentStream;
```
Comment (c43721) [inline src/lib/parsers/Args.ts:48]: Parser should have a default value

File: src/arguments/CoreBoolean.ts [hunk 2]
```diff
@@ -9,12 +9,13 @@ export class CoreArgument extends Argument<boolean> {
...
 	public run(parameter: string, context: { readonly truths?: string[]; falses?: readonly string[] } & Argument.Context): Argument.Result<boolean> {
 		const resolved = resolveBoolean(parameter, { truths: context.truths, falses: context.falses });
-		if (resolved.isOk()) return this.ok(resolved.unwrap());
-		return this.error({
-			parameter,
-			identifier: resolved.unwrapErr(),
-			message: 'The argument did not resolve to a boolean.',
-			context
-		});
+		return resolved.mapErr((identifier) =>
```
Comment (c43721) [inline src/arguments/CoreBoolean.ts:12]: `mapErrInto`? Was this not possible?

https://github.com/sapphiredev/framework/pull/468#discussion_r932544618

File: src/lib/structures/Command.ts [hunk 3]
```diff
@@ -88,7 +88,7 @@ export class Command<PreParseReturn = Args, O extends Command.Options = Command.
 		this.fullCategory = options.fullCategory ?? this.location.directories;
 		this.typing = options.typing ?? true;
 
-		this.lexer = new Lexure.Lexer({
+		this.lexer = new Lexer({
```
Comment (c43721) [inline src/lib/structures/Command.ts:91]: Or was it this that needed a default in the constructor / initial value? I forgot which nit and I can't find it on the review.

**Review by c43721 — CHANGES_REQUESTED** [review]
Thanks, a few comments

File: src/arguments/CoreBoolean.ts (hunk 2 above)
Comment (samfundev) [reply src/arguments/CoreBoolean.ts:12]: I didn't see that edit, I don't see any method called `mapErrInto` and I don't see another comment mentioning it. I'm not sure what that is.

File: src/lib/structures/Command.ts (hunk 3 above)
Comment (samfundev) [reply src/lib/structures/Command.ts:91]: This is one that needed the default and I can't do that. [This conversation](https://github.com/sapphiredev/framework/pull/468#discussion_r932626374) explains it.

Comment (c43721) [reply src/lib/structures/Command.ts:91]: @samfundev Wait for next result and update to that.
https://github.com/sapphiredev/utilities/pull/404/files#diff-63b98d6beb14a27a657df5479d7a87f5136e762688cc29c9f4543a96ac8a492aR64

Comment (c43721) [reply src/lib/structures/Command.ts:91]: wrong reply but this is for that map err bs

File: src/lib/parsers/Args.ts (hunk 1 above)
Comment (c43721) [reply src/lib/parsers/Args.ts:48]: https://github.com/sapphiredev/utilities/pull/404/files#diff-63b98d6beb14a27a657df5479d7a87f5136e762688cc29c9f4543a96ac8a492aR64

File: src/arguments/CoreBoolean.ts (hunk 2 above)
Comment (c43721) [reply src/arguments/CoreBoolean.ts:12]: https://github.com/sapphiredev/utilities/pull/404/files#diff-63b98d6beb14a27a657df5479d7a87f5136e762688cc29c9f4543a96ac8a492aR64

Comment (c43721) [reply src/arguments/CoreBoolean.ts:12]: @samfundev Wait for next result and update to that.
https://github.com/sapphiredev/utilities/pull/404/files#diff-63b98d6beb14a27a657df5479d7a87f5136e762688cc29c9f4543a96ac8a492aR64

**Comment by samfundev** [comment]
@c43721 Just to clarify, this PR is ready for review and the unit tests are failing because of 747be013ae95440d74a580c709dddabef2fc0c87.

**Review by c43721 — APPROVED** [review]

**Pull request merged**